source/templates/secrets.slim

```diff
@@ -6,17 +6,32 @@
 		article
 			ul.env-vars[order-by-drag models="envVarsOfSource.envVars"]
 				li[ng-repeat="envVar in envVarsOfSource.envVars"]
-					form[name="envVarForm"]
+					p.progress-indicator[progress-model="envVar.customProperties.progress"]
+					form[name="envVarForm" ng-if="!envVar.customProperties.progress.isInProgress" autocomplete="off" autocapitalize="none"]
 						.key-value-delete
-							input.key[type="text" name="key" ng-model="envVar.key" ng-model-options="{getterSetter: true}" custom-validators="{'key': Variable.isValidKey(envVar.key(), envVarsOfSource.envVars)}" placeholder="#{ data[:strings][:env_vars][:key_placeholder] }" skip-drag-order]
-							textarea.value[name="value" msd-elastic ng-model="envVar.value" ng-model-options="{getterSetter: true}" ng-trim="false" placeholder="#{ data[:strings][:env_vars][:value_placeholder] }" skip-drag-order]
-							button.delete-env-var.rebo.danger[type="button" ng-click="envVarsCtrl.removeEnvVarFromEnvVars(envVar, envVarsOfSource.envVars)"] = svg("cross")
+							input.key[type="text" name="key" ng-model="envVar.key" ng-model-options="{getterSetter: true}" ng-disabled="!envVar.isKeyChangeable" custom-validators="{'key': Variable.isValidKey(envVar.key(), envVarsOfSource.envVars)}" placeholder="#{ data[:strings][:env_vars][:key_placeholder] }" autocomplete="off" autocorrect="off" autocapitalize="none" spellcheck="false" skip-drag-order]
+							.value-with-actions.secret
+								textarea.value[name="value" msd-elastic ng-model="envVarsCtrl.valueGetterSetterForEnvVar(envVar)" ng-class="{'protected' : requestService.mode == 'website' && envVar.isProtected()}" ng-disabled="!envVar.shouldShowValue" ng-model-options="{getterSetter: true}" ng-trim="false" placeholder="#{ data[:strings][:env_vars][:value_placeholder] }" skip-drag-order]
+								.lock[ng-if="requestService.mode == 'website' && !envVar.isKeyChangeable && envVar.isProtected()"] = svg("main/menuicon-secrets")
+								eye-checkbox[ng-if="requestService.mode == 'website' && !envVar.isKeyChangeable && !envVar.isProtected()" ng-model="envVarsCtrl.valueVisibilityGetterSetterForEnvVar(envVar)" ng-model-options="{getterSetter: true}" checkbox-id-expression="'secret-env-var-show-value-checkbox-' + $index"]
+								.should-show-value[ng-if="requestService.mode == 'website'"]
```

Comment:
This needs the same `ng-if` condition as the `eye-checkbox`, currently it is appearing on small screen even after making it protected - throwing error when clicked.

![screen shot 2018-02-08 at 20 15 04](https://user-images.githubusercontent.com/8428264/35993175-c7f66ad8-0d0c-11e8-8399-a3679492ac26.png)